test/util/util.go

```diff
@@ -297,12 +315,47 @@ func AdmitWorkload(ctx context.Context, k8sClient client.Client, wl *kueue.Workl
 		return fmt.Errorf("admission can't be null")
 	}
 	wl.Status.Admission = admission
-	wl.Status.Conditions = []metav1.Condition{{
-		Type:               kueue.WorkloadAdmitted,
-		Status:             metav1.ConditionTrue,
-		LastTransitionTime: metav1.Now(),
-		Reason:             "AdmittedByTest",
-		Message:            fmt.Sprintf("Admitted by ClusterQueue %s", wl.Status.Admission.ClusterQueue),
-	}}
+	i := workload.FindConditionIndex(&wl.Status, kueue.WorkloadAdmitted)
+	if i == -1 {
+		wl.Status.Conditions = append(wl.Status.Conditions, metav1.Condition{
+			Type:               kueue.WorkloadAdmitted,
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: metav1.Now(),
+			Reason:             "AdmittedByTest",
+			Message:            fmt.Sprintf("Admitted by ClusterQueue %s", wl.Status.Admission.ClusterQueue),
+		})
+	} else {
+		wl.Status.Conditions[i] = metav1.Condition{
+			Type:               kueue.WorkloadAdmitted,
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: metav1.Now(),
+			Reason:             "AdmittedByTest",
+			Message:            fmt.Sprintf("Admitted by ClusterQueue %s", wl.Status.Admission.ClusterQueue),
+		}
+	}
```

Comment:
Probably, using the `apimeta.SetStatusCondition` is enough since the function checks whether the conditions already have the condition.

https://github.com/kubernetes/apimachinery/blob/61b874e0403b27353daf90f9789c762610731ab6/pkg/api/meta/conditions.go#L25-L55